src/Rules/Test.hs

```diff
@@ -50,33 +50,35 @@ testRules = do
             , builderEnvironment "TEST_HC" $ Ghc CompileHs Stage2
             , AddEnv "TEST_HC_OPTS" <$> runTestGhcFlags ]
 
-        makePath       <- builderPath $ Make ""
-        top            <- topDirectory
-        ghcPath        <- (top -/-) <$> builderPath (Ghc CompileHs Stage2)
-        ghcFlags       <- runTestGhcFlags
+        makePath        <- builderPath $ Make ""
+        top             <- topDirectory
+        ghcPath         <- (top -/-) <$> builderPath (Ghc CompileHs Stage2)
+        ghcFlags        <- runTestGhcFlags
+        checkPprPath    <- (top -/-) <$> needfile Stage1 checkPpr
+        annotationsPath <- (top -/-) <$> needfile Stage1 checkApiAnnotations
 
         -- Set environment variables for test's Makefile.
         liftIO $ do
             setEnv "MAKE" makePath
             setEnv "TEST_HC" ghcPath
             setEnv "TEST_HC_OPTS" ghcFlags
+            setEnv "CHECK_PPR" checkPprPath
+            setEnv "CHECK_API_ANNOTATIONS" annotationsPath 
 
         -- Execute the test target.
         buildWithCmdOptions env $ target (vanillaContext Stage2 compiler) RunTest [] []
 
 -- | Build extra programs required by testsuite
 needTestsuiteBuilders :: Action ()
 needTestsuiteBuilders = do
-    targets <- mapM (needfile Stage1) =<< testsuitePackages
+    targets        <- mapM (needfile Stage1) =<< testsuitePackages
+    binPath        <- stageBinPath Stage1
+    libPath        <- stageLibPath Stage1
+    iservPath      <- needfile Stage1 iserv 
+    runhaskellPath <- needfile Stage1 runGhc
```

Comment:
Shall `iserv` and `runGhc` be added to `testsuitePackages`?

Reply:
Yes, It we need iserv in testsuite packages as we need to build its profiling way. 
runGhc is already in stage1 and I dont't think we need to build any other way here. But we can add it just to be sure. What are your suggestions @snowleopard .

Reply:
The `testsuitePackages` should contain *all* packages required for the testsuite. That's what the name suggests!

Reply:
Aha, I see where the confusion can be coming from. The comment says: "Packages that are built only for the testsuite", but I think it should actually say "Packages that are required for the testsuite".

Reply:
Resolved. There are still few questions regarding this issue but I think I should create a separate issue for that.